builtin/mainmenu/tab_settings.lua

```diff
@@ -389,6 +437,18 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		core.setting_set("anisotropic_filter", "true")
 		ddhandled = true
 	end
+	
+	if fields["dd_resolution"] then
+		core.setting_set("screenW",
+			video_mode_to_option_w(fields["dd_resolution"]))
+		core.setting_set("screenH",
+			video_mode_to_option_h(fields["dd_resolution"]))
+		if fields["dd_resolution"] == default_label then
```

Comment:
No, this doesn't work in practice and it must recognize imperatively that the default_label has been selected to set those values.